### ISSUE TITLE
[Idea] Add option to show commit dialog beneath commit graph

### ISSUE THREAD

**Issue by mdonatas** [gitextensions/gitextensions#5535]
### Problem
Once Commit dialog is opened, you can't interact with commit graph and if you leave it open in another monitor clicking on main form reveals commit dialog forcing you to close it.

### Suggestion
(with some Paint .Net magic) I propose to have an option to place Commit dialog under commit graph next to Diff or File Tree tabs
![image](https://user-images.githubusercontent.com/483659/46599636-7fce7a00-caf0-11e8-8f89-8196228b66a5.png)

We could have an option to pin/unpin it which would switch this dialog between a full blown dialog as it is now and this docked state proposed in the screenshot. [ctrl+space] would work as expected too - focus Commit tab.

**Comment by vbjay** [comment]
It's been suggested before. That would kill the performance of the main
window.

**Comment by mdonatas** [comment]
Why would it do that? The contents of Commit dialog don't change based on what you do in Commit graph, there'd still be Refresh button to reload it and the form would be initialized only when selecting Commit tab.
What am I missing?

**Comment by vbjay** [comment]
See #564

**Comment by pmiossec** [comment]
It's like sourcetree does and I don't like it. The space to view diff and list of staged/unstaged files end up to be too narrow...

**Comment by mdonatas** [comment]
@vbjay the only comment about negative performance in that discussion was yours (dick-move) and the only reply to it was in essence that it wouldn't be a problem because of lazy loading.

**Comment by mdonatas** [comment]
@pmiossec  Like/not-like is not the same as good-idea/bad-idea ;) This would be an option to dock commit dialog and not replacement of current behavior. Also ctrl+space could first open commit form in its last state and switch between the two states when further invoking ctrl+space == great for small commits and large commits as transformation is quickly invoked.

**Comment by mdonatas** [comment]
Duplicate of #564

**Comment by RussKie** [comment]
I'm going to have this discussion open (because it is newer and easier to find) and close the other one.
I'm happy to return to this discussion after v3 is released.

**Comment by mstv** [comment]
In general, I want to be able to minimize an application regardless whether it shows a modal dialog or not.
I expect that minimizing the modal dialog hides the whole application.
I expect that the taskbar context menu item `Close Window` closes the _application_ or at least the active window (which doesn't have any effect on the Commit dialog currently).

Most of the time, I use windows maximized on one of the displays.
I like the idea of Ctrl+Space switching between modal Commit dialog (default) and docked operation.
I would appreciate to use it when I want to check something in previous commits while reviewing the current commit and when I want to copy _parts_ of a previous commit message.

The commit message should be stored always if modified by the user, especially also if it is a merge message -- on close of the Commit dialog and if docked, on switch of the repository.

**Comment by drewnoakes** [comment]
@RussKie #564 is not a duplicate. It just suggests making the window modeless, so you can tab between form commit/browse. It's a different proposal, and likely easier to implement. In fact it's not even mutually exclusive. Personally, I would rather see it made modeless than docked beneath the graph. The modal window breaks many desktop paradigms and features.

**Comment by johndog** [comment]
Hat tip to #564.

My opinion:

- First order of business is to make the dialog modeless.  This addresses most of the challenges of running commit UI and main UI concurrently.
- Second order is to make the dialog dockable.

I would rarely ever dock it, personally. I would want to use my window manager, multiple monitors, etc, to make sure that the UI has enough space depending on the environment.  As others have commented, the docked UI is usually going to be too small.

Today, I launch two separate instances of GitEx: one to run the main window, the other so that I can bring up the modal commit UI.  This lets me quickly switch back and forth between the commit UI and the graph -- the graph is essential to helping me compose commit messages, especially !fixups.  This is still rather painful, as I have to keep track of which instance has the blocking dialog and which doesn't, so I seek precisely the same experience with one instance of GitEx by way of modeless commit UI.  This should have no performance implications, since both UIs are refreshed on demand and independently. 

Performance would be maintained in a docking scenario by retaining the same behavior: the windows continue to be refreshed on-demand and independently.  It would help if the UI could provide some "stale-ness" indicator.  Even in current builds such an indicator would be handy for the main window, which often looks up-to-date but is completely wrong about the current branch, current commit, etc.  But we don't have a staleness indicator today, so I don't consider it a blocker for modeless commit UI, though it might be more important for docked UI.

**Comment by johndog** [comment]
> ... make the dialog modeless.  This addresses most of the challenges of running commit UI and main UI concurrently.

I just realized that describing this as "_most_ of the challenges" is wildly speculative.  I can imagine a tremendous amount of refactoring might be required to enable the same Commit UI logic to operate as a modeless window or as a docked window. 

It may even be the case that the only thing of value that a modeless implementation would accrue to a dockable implementation is to allow people try it out and expose the flaws in the experience of having Commit UI and main window UI concurrently accessible.  That'd still be a good thing to have in hand before starting work on a docked experience.

**Comment by gluttony38** [comment]
In my personal opinion, https://github.com/gitextensions/gitextensions/issues/564 would have been better to implement, anyway, there is a workaround, at least under Windows, that is to open folder in file explorer, right click and select "GitExt Commit...", this will open the commit dialog independently to any other Git Extensions window opened.

**Comment by gerhardol** [comment]
More and more functionality in FormCommit is available in the diff tab for artificial commits
When #7825 is merged I will only have to open FormCommit to enter the commit message

**Comment by githubkusi** [comment]
I don't have a strong opinion for either #5535 (this) or #564, but this is currently quite a nuisance: I often have the commit window next to an application window (Matlab in my case) since I need the application in order to write the commit message. This is not possible if the main GE window is in the way, and there is no way to move the main GE window away.

+1 for modeless commit window or commit tab
